src/components/engagement/engagement.service.ts

```diff
@@ -793,7 +799,7 @@ export class EngagementService {
     );
     await this.eventBus.publish(engagementUpdatedEvent);
 
-    return engagementUpdatedEvent.updated as LanguageEngagement;
+    return (await this.readOne(input.id, session)) as LanguageEngagement;
```

Comment:
Instead of calling readOne again you can just replace the `updated` object on the event which will be used in the return previously here.

```tsx
event.updated = {
   ...event.updated,
  statusModifiedAt: {...}
};
```